### ISSUE TITLE
[a11y][iOS] Nested accessibility items not accessible

### ISSUE THREAD

**Issue by cihan-m** [facebook/react-native#27147]
Hello,

This is basically a repeat of #24515. It seems like that issue is locked though, so creating a new one.

React Native version: 0.61.2
```
System:
    OS: macOS 10.15.1
    CPU: (8) x64 Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
    Memory: 1.32 GB / 16.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 8.11.3 - /usr/local/bin/node
    npm: 6.4.1 - /usr/local/bin/npm
  SDKs:
    iOS SDK:
      Platforms: iOS 13.1, DriverKit 19.0, macOS 10.15, tvOS 13.0, watchOS 6.0
  IDEs:
    Android Studio: 3.5 AI-191.8026.42.35.5900203
    Xcode: 11.1/11A1027 - /usr/bin/xcodebuild
  npmPackages:
    react: 16.9.0 => 16.9.0 
    react-native: 0.61.2 => 0.61.2 
  npmGlobalPackages:
    react-native-cli: 2.0.1
```

## Steps To Reproduce

1. Use nested Text components for text inline with hyperlinks, or go to this Snack: https://snack.expo.io/@heehee.its.cindy/snack-to-reproduce-nested-accessibility-issues
2. Turn on Voiceover, swipe around the app
3. See that there is no way for a user using Voiceover to access the hyperlink to click it

Describe what you expected to happen:
Voiceover user should be able to access the entirety of the text, and the hyperlink (so both the parent Text component and nested Text components). There were two strategies proposed by @sahrens in #24515, can we open that issue back up to continue that discussion?

**Comment by CoryWritesCode** [comment]
I may be asking the wrong question but why would you nest `<Text />` within a `<Text />`?

**Comment by cihan-m** [comment]
@CoryWritesCode this is how we’ve been creating text all in one line where different parts of the text needs different styling. For example, if one part of a sentence needs a hyperlink.

Wrapping with a View even with flexWrap “wrap” has some visual issues, unless maybe we create a separate child Text component for every word. So there is a sort of a clunky workaround, however it seems like native iOS and Flutter have a nicer way of handling nested accessibility, that would be nice to have in React Native as well.

**Comment by CoryWritesCode** [comment]
I can see that @cihan-m. Makes sense as RN doesn't have a `<Span />` type component.
